kinematics_interface/include/kinematics_interface/kinematics_interface.hpp

```diff
@@ -110,6 +122,10 @@ class KinematicsInterface
   bool calculate_jacobian(
     const std::vector<double> & joint_pos_vec, const std::string & link_name,
     Eigen::Matrix<double, 6, Eigen::Dynamic> & jacobian);
+
+  bool calculate_frame_difference(
+    std::vector<double> & x_a_vec, std::vector<double> & x_b_vec, double dt,
+    std::vector<double> & delta_x_vec);
```

Comment:
Why not imbibe the size constraints?

```suggestion
    std::array<double, 7> & x_a_vec, std::array<double, 7> & x_b_vec, double dt,
    std::array<double, 6> & delta_x_vec);
```